src/components/colorscale/defaults.js

```diff
@@ -33,9 +33,14 @@ module.exports = function colorScaleDefaults(traceIn, traceOut, layout, coerce,
     var minIn = containerIn[cLetter + 'min'];
     var maxIn = containerIn[cLetter + 'max'];
     var validMinMax = isNumeric(minIn) && isNumeric(maxIn) && (minIn < maxIn);
-    coerce(prefix + cLetter + 'auto', !validMinMax);
-    coerce(prefix + cLetter + 'min');
-    coerce(prefix + cLetter + 'max');
+    var auto = coerce(prefix + cLetter + 'auto', !validMinMax);
+
+    if(auto) {
+        coerce(prefix + cLetter + 'mid');
+    } else {
+        coerce(prefix + cLetter + 'min');
```

Comment:
this line and the next used to always run, but now only conditionally... might this break anything? I'm not sure about the implication of not calling `coerce()`.

Reply:
No tests are failing because of it.

Reply:
ok

Reply:
There is a test failure that mentions `coerce`:

> Test surface supplyDefaults should coerce 'c' attributes with 'c' values regardless of `'z' if 'c' is present FAILED

Reply:
Ah good eye.

Well `cmin` and `cmax` have no effect

https://github.com/plotly/plotly.js/blob/bb1239a31a56d72ec664091ca7e1c0b42adb8b2f/src/components/colorscale/calc.js#L31-L37

when `cauto` is true.

Reply:
I'm easy. Like I said, I don't know the impact of calling `coerce` and ignoring the attr vs not calling `coerce` :)

Reply:
Calling the behaviour correct and adapting the test in https://github.com/plotly/plotly.js/pull/3549/commits/ff6781226bf5717efdcf0c9041b9e5f0b6e7abc6